scene/2d/path_2d.cpp

```diff
@@ -87,81 +215,231 @@ bool Path2D::_edit_is_selected_on_click(const Point2 &p_point, double p_toleranc
 
 void Path2D::_notification(int p_what) {
 	switch (p_what) {
-		// Draw the curve if path debugging is enabled.
+		case NOTIFICATION_ENTER_TREE: {
+			PathDebug2D::add_update_callback(this, callable_mp(this, &Path2D::_on_debug_global_changed));
+			_debug_create();
+			queue_redraw();
+		} break;
+
+		case NOTIFICATION_EXIT_TREE: {
+			PathDebug2D::remove_update_callback(this);
+			_debug_free();
+		} break;
+
 		case NOTIFICATION_DRAW: {
-			if (curve.is_null()) {
-				break;
-			}
+			_debug_update();
+		} break;
+	}
+}
 
-			if (!Engine::get_singleton()->is_editor_hint() && !get_tree()->is_debugging_paths_hint()) {
-				return;
-			}
+void Path2D::_on_debug_global_changed() {
+	if (_emitting_debug_changed) {
+		return;
+	}
+	_emitting_debug_changed = true;
+	callable_mp(this, &Path2D::_emit_debug_changed_deferred).call_deferred();
+}
 
-			if (curve->get_point_count() < 2) {
-				return;
-			}
+void Path2D::_emit_debug_changed_deferred() {
```

Comment:
This method's name is also super misleading. It does not emit anything.

Reply:
agree, it is remnant. That function originally did something else hence the naming.